File: JavaBackend/MistyBackend/src/model/GleimMenu.java
package model;

import java.util.ArrayList;

public class GleimMenu {

    private ArrayList<GleimMenu> subMenuList = new ArrayList<>();
    private String name;
    private String audioFilePath;
    private ArrayList<String> commandList = new ArrayList<>();
    private String serialCommand;
    private ArrayList<String> returnCommandList = new ArrayList<>();

    public GleimMenu(String name){
        this.name=name;
    }

    public ArrayList<String> getSubMenuCommandList(){
        ArrayList<String> array = new ArrayList<>();
        for(GleimMenu menuElement : subMenuList){
            for(String command : menuElement.getCommandList()){
                array.add(command);
            }
        }
        return  array;
    }

    public void addSubMenu(GleimMenu gleimMenu){
        subMenuList.add(gleimMenu);
    }

    public void addCommand(String command){
        commandList.add(command);
    }

    public void addReturnCommand(String command){
        returnCommandList.add(command);
    }

    //-----------Generated Getter Setter------------

    public ArrayList<GleimMenu> getSubMenuList() {
        return subMenuList;
    }

    public void setSubMenuList(ArrayList<GleimMenu> subMenuList) {
        this.subMenuList = subMenuList;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAudioFilePath() {
        return audioFilePath;
    }

    public void setAudioFilePath(String audioFilePath) {
        this.audioFilePath = audioFilePath;
    }

    public ArrayList<String> getCommandList() {
        return commandList;
    }

    public void setCommandList(ArrayList<String> commandList) {
        this.commandList = commandList;
    }

    public String getSerialCommand() {
        return serialCommand;
    }

    public void setSerialCommand(String serialCommand) {
        this.serialCommand = serialCommand;
    }

    public ArrayList<String> getReturnCommandList() {
        return returnCommandList;
    }

    public void setReturnCommandList(ArrayList<String> returnCommandList) {
        this.returnCommandList = returnCommandList;
    }
}

File: MistyTest/GleimMenu.swift
//
//  GleimMenuList.swift
//  MistyTest
//
//  Created by Jahresprojekt2017/18 on 08.04.18.
//  Copyright © 2018 Ivo Max Muellner. All rights reserved.
//

import Foundation

class GleimMenu{
    
    private var subMenuList = [GleimMenu]()
    private var name = ""
    private var audioFilePath = ""
    private var commandList = [String]()
    private var serialCommand = ""
    private var returnCommandList = [String]()
    
    init(name: String, subMenu: Array<GleimMenu>, audioFilePath: String, commandList: Array<String>, serialCommand: String, returnCommandList: Array<String>){
        self.name=name
        self.subMenuList=subMenu
        self.audioFilePath=audioFilePath
        self.commandList=commandList
        self.serialCommand=serialCommand
        self.returnCommandList=returnCommandList
    }
    
    init(name: String){
        self.name=name
    }
    
    init(){
        
    }
    
    public func getSubMenuList() -> Array<GleimMenu> {
        return subMenuList
    }
    
    public func getName() -> String {
        return name
    }
    
    public func getAudioFilePath() -> String {
        return audioFilePath
    }
    
    public func getOwnCommandList() -> Array<String> {
        return commandList
    }
    
    public func getSubMenuCommandList() -> Array<String> {
        var array = [String]()
        for menuElement in subMenuList {
            for command in menuElement.getOwnCommandList() {
                array.append(command)
            }
        }
        return array
    }
    
    public func getSerialCommand() -> String {
        return serialCommand
    }
    
    public func getReturnCommandList() -> Array<String> {
        return returnCommandList
    }
    
    public func setName(name: String) {
        self.name=name
    }
    
    public func setAudioFilePath(path: String){
        audioFilePath=path
    }
    
    public func setCommandList(list: Array<String>){
        commandList=list
    }
    
    public func addCommand(command: String){
        commandList.append(command)
    }
    
    public func addSubMenuElement(element: GleimMenu){
        subMenuList.append(element)
    }
    
    public func setSerialCommand(serialCommand: String) {
        self.serialCommand=serialCommand
    }
    
    public func setReturnCommandList(returnCommandList: Array<String>) {
        self.returnCommandList=returnCommandList
    }
    
    public func addReturnCommand(command: String){
        returnCommandList.append(command)
    }
    
}
